fuel/schemes.py

```diff
@@ -58,9 +58,25 @@ class BatchScheme(IterationScheme):
     to be kept in memory, like MNIST) we can provide slices or lists of
     labels to the dataset.
 
+    Parameters
+    ----------
+    examples : int or list
+        Defines which examples from the dataset are iterated.
+        If list, its items are the indices of examples.
+        If an integer, it will use that many examples from the beginning
+        of the dataset, i.e. it is interpreted as range(examples)
+    batch_size : int
+        The request iterator will return slices or list of indices in
+        batches of size `batch_size` until the end of `examples` is
+        reached.
+        Note that this means that the last batch size returned could be
+        smaller than `batch_size`. If you want to ensure all batches are
+        of equal size, then ensure len(`examples`) or `examples` is a
+        multiple of `batch_size`.
+
     """
-    def __init__(self, num_examples, batch_size):
-        self.num_examples = num_examples
+    def __init__(self, examples, batch_size):
+        self.indices = xrange(examples) if type(examples) is int else examples
```

Comment:
We should probably allow for duck typing instead of checking explicitly if the type is `int`. There's two options I think: Either use
```python
from numbers import Integral
isinstance(examples, Integral)
```
or
```
from collections import Iterable
not isinstance(x, Iterable)
```